src/game/client/draw/drawable.cpp

```diff
@@ -985,3 +1002,826 @@ void Drawable::Draw_UI_Text()
+            return (DrawableIconType)i;
+        }
+    }
+
+    return ICON_INVALID;
+}
+
+void Drawable::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 7;
+    xfer->xferVersion(&version, 7);
+
+    if (xfer->Get_Mode() == XFER_LOAD && m_ambientSound != nullptr) {
+        g_theAudio->Kill_Event_Immediately(m_ambientSound->m_event.Get_Playing_Handle());
+        m_ambientSound->Delete_Instance();
+        m_ambientSound = nullptr;
+    }
+
+    DrawableID id = Get_ID();
+    xfer->xferDrawableID(&id);
+    Set_ID(id);
+
+    if (version >= 2) {
+        m_conditionState.Xfer(xfer);
+
+        if (xfer->Get_Mode() == XFER_LOAD) {
+            Replace_Model_Condition_Flags(m_conditionState, true);
+        }
+    }
+
+    if (version >= 3) {
+        if (version < 5) {
+            Coord3D pos = *Get_Position();
+            xfer->xferCoord3D(&pos);
+            Set_Position(&pos);
+            float angle = Get_Orientation();
+            xfer->xferReal(&angle);
+            Set_Orientation(angle);
+        } else {
+            Matrix3D m = *Get_Transform_Matrix();
+            xfer->xferMatrix3D(&m);
+            Set_Transform_Matrix(&m);
+        }
+    }
+
+    bool selection_color = m_selectionColorEnvelope != nullptr;
+    xfer->xferBool(&selection_color);
+
+    if (selection_color) {
+        if (m_selectionColorEnvelope == nullptr) {
+            m_selectionColorEnvelope = new TintEnvelope();
+        }
+
+        xfer->xferSnapshot(m_selectionColorEnvelope);
+    }
+
+    bool tint_color = m_tintColorEnvelope != nullptr;
+    xfer->xferBool(&tint_color);
+
+    if (tint_color) {
+        if (m_tintColorEnvelope == nullptr) {
+            m_tintColorEnvelope = new TintEnvelope();
+        }
+
+        xfer->xferSnapshot(m_tintColorEnvelope);
+    }
+
+    TerrainDecalType decal = Get_Terrain_Decal();
+    xfer->xferUser(&decal, sizeof(decal));
+
+    if (xfer->Get_Mode() == XFER_LOAD) {
+        Set_Terrain_Decal(decal);
+    }
+
+    xfer->xferReal(&m_opacity);
+    xfer->xferReal(&m_effectiveOpacity1);
+    xfer->xferReal(&m_effectiveOpacity2);
+    xfer->xferReal(&m_terrainDecalFadeTarget1);
+    xfer->xferReal(&m_terrainDecalFadeTarget2);
+    xfer->xferReal(&m_terrainDecalOpacity);
+
+    ObjectID object_id;
+    if (m_object != nullptr) {
+        object_id = m_object->Get_ID();
+    } else {
+        object_id = OBJECT_UNK;
+    }
+
+    xfer->xferObjectID(&object_id);
+
+    if (xfer->Get_Mode() == XFER_LOAD) {
+        if (m_object != nullptr) {
+            captainslog_relassert(object_id == m_object->Get_ID(),
+                6,
+                "Drawable::xfer - Drawable '%s' is attached to wrong object '%s'",
+                Get_Template()->Get_Name().Str(),
+                m_object->Get_Template()->Get_Name().Str());
+        } else if (object_id != OBJECT_UNK) {
+            Object *object = g_theGameLogic->Find_Object_By_ID(object_id);
+            captainslog_relassert(object_id == OBJECT_UNK,
```

Comment:
This assert will hit 100% currently.

Reply:
No its doing what its supposed to do.
It wont hit 100% (otherwise it would have hit in my testing).

Reply:
Did you put a breakpoint to see if the code hits?

Look, it enters condition:
`if (object_id != OBJECT_UNK)`
so it expects object id to not be UNK.

Then it asserts on
`object_id == OBJECT_UNK`

So not it expects id to be UNK.

Either I a went full retard or this is wrong.

Reply:
The code says "if object_id != OBJECT_UNK then "assert if object_id != OBJECT_UNK" which is what its supposed to be doing.
The assert is meant to trip anytime object_id != OBJECT_UNK (but that whole thing will only run if m_object is nullptr)